samza-core/src/main/scala/org/apache/samza/job/local/ThreadJobFactory.scala

```diff
@@ -54,18 +52,31 @@ class ThreadJobFactory extends StreamJobFactory with Logging {
       case _ => None
     }
 
+    val containerListener = new SamzaContainerListener {
+      override def onContainerFailed(t: Throwable): Unit = {
+        throw t
```

Comment:
Can't ThreadJob be run w/o the ApplicationRunner (i.e. with the low-level APIs)?